### ISSUE TITLE
Regression: ConfigError on flat/recommended-typescript-error since 54.6.0

### ISSUE THREAD

**Issue by TomCorvus** [gajus/eslint-plugin-jsdoc#1474]
Hello,

## Expected behavior

That the linter works like on 54.5.0 version.

## Actual behavior

Since 54.6.0 version, I have an error when I run Eslint with JSDoc.
`ConfigError: Config "jsdoc/flat/recommended-typescript-error": Key "plugins": Key "jsdoc": Expected an object.`
If I log the config, it prints:
```
{
  name: 'jsdoc/flat/recommended-typescript-error',
  plugins: { jsdoc: [Function: index] { configs: [Object], rules: [Object] } },
  rules: {
    'jsdoc/check-access': 'error',
```

## ESLint Config

```js
import js from '@eslint/js'
import ts from 'typescript-eslint'
import jsdoc from 'eslint-plugin-jsdoc'
import react from 'eslint-plugin-react'

const config = ts.config(
	js.configs.recommended,
	ts.configs.recommended,
	jsdoc.configs['flat/recommended-typescript-error'],
	{
		files: [
			'**/*.js',
			'**/*.jsx',
			'**/*.ts',
			'**/*.tsx',
			'**/*.mts',
			'**/*.cts',
		],
		plugins: {
			react,
		},
		languageOptions: {
			ecmaVersion: 2022,
			sourceType: 'module',
			parser: ts.parser,
			globals: {
				browser: true,
				es6: true,
				jest: true,
				node: true,
			},
		},
	},
)

export default config
```

## ESLint sample

```js
eslint --fix
```

## Environment

- Node version: 22.19.0
- ESLint version 9.35.0
- `eslint-plugin-jsdoc` version: 54.6.0
